### ISSUE TITLE
Python 3 support

### ISSUE THREAD

**Issue by ejeschke** [karimbahgat/PyAgg#6]
Just found this project and it looks very promising.

I am using the `aggdraw` module for [one of my projects](https://github.com/ejeschke/ginga) and I have fixed a couple bugs (path drawing) and added Python 3 support.  You can find my version of `aggdraw` *[here](https://github.com/ejeschke/aggdraw)*.

I tried running a couple of your tests using python 3, but it looks like the code is still python 2 only (old style print statements, etc.).  You might consider upgrading the code to be python-2 or 3 compatible.  I found the `modernize` (aka python-modernize) package to be very useful for converting the files in my project, requiring only minimal fix ups afterwards.  It uses the `six` module to achieve compatibility.

**Comment by karimbahgat** [comment]
You're right I havent yet thought about py3 support, but its probably time. Will look into modernize when i can, thanks. And your project looks really interesting as well! 

What great news to hear you made aggdraw py3 compatible! I was actually wanting to do that myself at some point, to avoid aggdraw dying out with py2. 

In fact, Ive noticed several have forked aggdraw and made many separate improvements. So one thing I've thought about is creating an official fork of aggdraw, and merging as many of these as possible. Sort of like what Pillow did for PIL, it could be called maybe Pillow-Case, as a complement to Pillow. Ive already started and merged a few such as adding line join and cap support. Your py3 support would be a great addition there. Would love to have another aggdrawer on the team that knows how to tweak the source code?

**Comment by ejeschke** [comment]
> Ive already started and merged a few such as adding line join and cap support. Your py3 support would be a great addition there. Would love to have another aggdrawer on the team that knows how to tweak the source code?

Please take what you want from the fork, it would be good to keep `aggdraw` moving forward, even if slowly.  I'm not an expert on the `aggdraw` code by any means, but it seems pretty well designed and somewhat straightforward to modify.

**Comment by karimbahgat** [comment]
I finally got around to it, and made a GitHub organization and [new library with just a slightly tweaked name "aggdrawer"](https://github.com/python-aggdrawer/aggdrawer), that can hopefully become the new home of future aggdraw improvements/forks. I started with the latest stable version and added your py3 support and path fix. 

If you would like i could make you a comaintainer, otherwise feel free to just submit PRs if you make any future changes.

**Comment by djhoese** [comment]
@karimbahgat I have taken over the `aggdraw` project (just found this repository from your pycrs project) and have included the changes from @ejeschke. My group (pytroll) also got effbot (the original author) to give us permissions to the PyPI project so we've been able to make real releases. Also `aggdraw` is now available on `conda-forge`.

https://github.com/pytroll/aggdraw
https://github.com/conda-forge/aggdraw-feedstock

**Comment by djhoese** [comment]
Feel free to make any pull requests and I'll merge them. I'm a little hesitant to update the version of agg library used without some heavy unit tests being added.

**Comment by karimbahgat** [comment]
That is really great news @djhoese, especially that it is group-coordinated and you have permission to the official pypi project! Then I'll definitely drop my previous [aggdrawer project](https://github.com/python-aggdrawer/aggdrawer), and do any future changes to the the pytroll repo instead. 

Aggdraw is sufficiently lightweight and fast compared to Python rendering libs like matplotlib and cairo that it is definitely worth maintaining and expanding, and there is so much additional functionality that can be incorporated from the underlaying agg library, incl such things as gradients, line patterns, ++.

**Comment by karimbahgat** [comment]
Added some issues to your fork for things I would like to see in the future, and/or as points of discussion.

**Comment by djhoese** [comment]
@karimbahgat Much appreciated. If we keep this back and forth issue marathon going we're going to get a lot of work done 😉 . I think you can close this issue by the way as the "official" aggdraw on PyPI now supports python 3.

Edit: Nevermind, I guess aggdraw supports python 3 but your code in pyagg may not.

**Comment by karimbahgat** [comment]
@djhoese Hahah very true, it's always more motivating/effective when someone else is waiting for your answers/code. 

And you're right, pyagg is just a wrapper around aggdraw, which still needs to upgrade to py3 at some point. At that point I would also need to update the underlaying aggdraw version (since these are provided precompiled) to your newest Py3-supporting official fork.
